File: backend/app/schemas/task_profile_log.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime

class TaskProfileLogOut(BaseModel):
    id: int
    task_profile_id: Optional[int] = None
    title: str
    detail: Optional[str] = None
    fname: str
    lname: str
    phone_num: Optional[str] = None
    email: Optional[str] = None
    birth_date: Optional[date] = None
    status: str
    action: str
    action_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/services/taskProfileLog.ts
import axios from 'axios';

const BASE_URL = 'http://localhost:8080';

export interface TaskProfileLog {
  id: number;
  task_profile_id: number | null;
  title: string;
  detail: string | null;
  fname: string;
  lname: string;
  phone_num: string | null;
  email: string | null;
  birth_date: string | null;
  status: 'active' | 'inactive' | 'done';
  action: 'create' | 'update' | 'delete';
  action_at: string;
}

export const getTaskProfileLogs = async (): Promise<TaskProfileLog[]> => {
  const response = await axios.get(`${BASE_URL}/taskprofilelog`);
  return response.data;
};